### PR TITLE
Factor out the images.meta loading code from YoloExtractor

### PR DESCRIPTION
**Pull request by IRDonch**
### Summary
It looks like the same thing will be needed for Open Images, so I'm
moving it to a common module.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [x] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: datumaro/util/image.py [hunk 1]
```diff
@@ -380,3 +382,32 @@ def save(self, path):
...
+
+    if not osp.isfile(image_meta_path):
+        raise Exception("Can't read image meta file '%s'" % image_meta_path)
+
+    image_meta = {}
+
+    with open(image_meta_path, encoding='utf-8') as f:
+        for line in f:
+            image_name, h, w = line.strip().rsplit(maxsplit=2)
+            image_meta[image_name] = (int(h), int(w))
```
Comment (zhiltsov-max) [inline datumaro/util/image.py:411]: Probably, we need to skip empty (and maybe commented) lines here too.
```suggestion
            line = line.strip()
            if not line or line.startswith('#'):
                continue
            
            image_name, h, w = line.strip().rsplit(maxsplit=2)
            image_meta[image_name] = (int(h), int(w))
```

Comment (IRDonch) [reply datumaro/util/image.py]: I implemented an approach based on `shlex.split` - that handles comments and allows other fields to be added in the future. What do you think?

Comment (zhiltsov-max) [reply datumaro/util/image.py]: Maybe it is a good solution, but I find it hard to understand the grammar it describes from the official description (both in POSIX and non-POSIX modes), so I'm having troubles with predicting the possible consequences.

Comment (IRDonch) [reply datumaro/util/image.py]: It's just (very simplified) shell syntax. The main consequence is that file names with spaces in them have to be quoted.

Comment (zhiltsov-max) [reply datumaro/util/image.py]: Consider updating few other formats to use `shlex`, such syntax is used in few extra dataset files from datumaro - VOC, CamVid and several others.

**Review by zhiltsov-max — APPROVED** [review]

File: datumaro/util/image.py (hunk 1 above)
Comment (IRDonch) [reply datumaro/util/image.py]: I will take a look later, but right now I want to get back to Open Images.

FWIW, I found four likely candidates for updating:

```console
$ rg -Fl "line[0] == '#'"
datumaro/plugins/cityscapes_format.py
datumaro/plugins/voc_format/format.py
datumaro/plugins/camvid_format.py
datumaro/plugins/kitti_format/format.py
```

Not sure about LFW.

**Pull request merged**